app/javascript/components/Profile/ChangePassword.tsx

```diff
@@ -0,0 +1,159 @@
+import React from 'react';
+import { Form, InputOnChangeData, Message, Segment, Header } from 'semantic-ui-react';
+import { inject, observer } from 'mobx-react';
+import { computed } from 'mobx';
+import SessionStore, { RequestState } from '../../stores/session_store';
+
+
+interface InjectedProps {
+  sessionStore: SessionStore;
+}
+
+@inject('sessionStore')
+@observer
+export default class ChangePassword extends React.Component {
+  state = {
+    isSafe: true,
```

Comment:
shouldn't we initially assume that these values are both set to `false`?

Reply:
then you would have an initial error state - normally you enter the passwords and either on moving focus out of the current input or submitting the form, the check is performed... that's why `isSafe` is true initially... how would you model that?